### ISSUE TITLE
Timeout configuration

### ISSUE THREAD

**Issue by thesmart** [sendgrid/sendgrid-ruby#172]
#### Issue Summary

Please add configuration options for connection open and read timeouts.  The `Net::HTTP` default of 60 seconds is far too long for a server configuration.

#### Technical details:

* sendgrid-ruby Version: master
* Ruby Version: 2.3.1

**Comment by mbernier** [comment]
@thesmart it looks like https://github.com/sendgrid/ruby-http-client/pull/20 will solve this issue! 

@jrectenwald - Would you be up for adding documentation to this repo to explain how to change the http_options in an API call?

**Comment by xxswingxx** [comment]
Was this gem updated to support `http_options`? Looks like only the `ruby_http_client` was updated.

**Comment by NeoPhi** [comment]
Looks like the upstream `ruby_http_client` hasn't released a new version to pull in.
In the meantime, if you don't mind monkey patching, this worked for us:
```
module SendGrid
  class Client
    def make_request(http, request)
      http.open_timeout = 10
      http.read_timeout = 10
      response = http.request(request)
      Response.new(response)
    end
  end
end
```

**Comment by thinkingserious** [comment]
Hello @xxswingxx,

The next release will include the updated `ruby_http_client`. I don't have a specific time frame, but there is a release planned.

@NeoPhi,

Thanks for providing the work around!

With Best Regards,

Elmer

**Comment by gareth** [comment]
I think this is closed by #405 🎉
